exec/java-exec/src/test/java/org/apache/drill/exec/expr/fn/impl/TestStringFunctions.java

```diff
@@ -113,6 +113,75 @@ public void testSplitPart() throws Exception {
         .go();
   }
 
+  @Test
+  public void testSplitPartStartEnd() throws Exception {
+    testBuilder()
+      .sqlQuery("select split_part(a, '~@~', 1, 2) res1 from (values('abc~@~def~@~ghi'), ('qwe~@~rty~@~uio')) as t(a)")
+      .ordered()
+      .baselineColumns("res1")
+      .baselineValues("abc~@~def")
+      .baselineValues("qwe~@~rty")
+      .go();
+
+    testBuilder()
+      .sqlQuery("select split_part(a, '~@~', 2, 3) res1 from (values('abc~@~def~@~ghi'), ('qwe~@~rty~@~uio')) as t(a)")
+      .ordered()
+      .baselineColumns("res1")
+      .baselineValues("def~@~ghi")
+      .baselineValues("rty~@~uio")
+      .go();
+
+    // invalid index
+    boolean expectedErrorEncountered;
```

Comment:
Could you break some of these out into separate unit tests?  Particularly, the failing tests?
Here's an example if you need some reference code:

https://github.com/apache/drill/blob/9a9005aa5eb41e07c11a935521f8635f7b79b6c5/contrib/format-excel/src/test/java/org/apache/drill/exec/store/excel/TestExcelFormat.java#L204-L217